src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,813 @@
+        assertExport(null, ExportType.DAILY, uploadId2, 0, false, false, false);
+    }
+
+    @Test
+    public void testDailyAllStudies() throws Exception {
+        // do NOT run this test on prod env
+        if (isProduction(env)) return;
+
+        assertExport(null, ExportType.DAILY, uploadId, 0, true, false, false);
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // check again
+        assertExport(null, ExportType.DAILY, uploadId2, 1, true, false, false);
+    }
+
+    @Test
+    public void testHourlyOnlyTestStudy() throws Exception {
+        // modify uploadedOn field in record table to a fake datetime earlier than 1 hour ago
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        ddbRecordTable.updateItem("id", uploadStatus.getRecord().getId(),
+                new AttributeUpdate("uploadedOn").put(DateTime.now().minusHours(2).getMillis()));
+
+        assertExport(null, ExportType.HOURLY, uploadId, 0, false, false, true);
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // check again
+        assertExport(null, ExportType.HOURLY, uploadId2, 0, false, false, false);
+    }
+
+    @Test
+    public void testHourlyAllStudies() throws Exception {
+        assertExport(null, ExportType.HOURLY, uploadId, 0, true, false, false);
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // check again
+        assertExport(null, ExportType.HOURLY, uploadId2, 1, true, false, false);
+    }
+
+    @Test
+    public void testDailyIgnoreLastExportDateTime() throws Exception {
+        // upload a test record to create last export date time field
+        assertExport(null, ExportType.DAILY, uploadId, 0, false, false, false);
+
+        // modify uploadedOn field in record table to a fake datetime earlier than 24 hours ago
+        // so that the first upload will not be exported again by exporter
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        ddbRecordTable.updateItem("id", uploadStatus.getRecord().getId(),
+                new AttributeUpdate("uploadedOn").put(DateTime.now().minusDays(1).minusHours(1).getMillis()));
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
```

Comment:
oh yeah i probably needs to modify the uploadedOn for this upload2 before lastexportdatetime